crates/bevy_gltf/src/loader/extensions/khr_materials_specular.rs

```diff
@@ -41,11 +39,15 @@ pub(crate) struct SpecularExtension {
 }
 
 impl SpecularExtension {
-    pub(crate) fn parse(
-        _load_context: &mut LoadContext,
-        _document: &Document,
-        material: &Material,
-    ) -> Option<Self> {
+    #[expect(
+        clippy::allow_attributes,
+        reason = "`unused_variables` is not always linted"
+    )]
+    #[allow(
+        unused_variables,
+        reason = "Depending on what features are used to compile this crate, certain parameters may end up unused."
+    )]
```

Comment:
not a fan of this combination of expect / allow, do we know the set of features?

Reply:
This is copied from the other extensions, which also use the same pattern. In this particular case, the relevant feature is pbr_specular_textures.

Note that previously we got away without these expect/allow by just naming the vars with a leading underscore to allow them to be unused, but then using them anyway, which I think is worse IMO.

If we want to fix this, let me know what you'd prefer - keep in mind if we conditionally pass these arguments, we'll have to change the call sites with cfg. Either way, I'd rather fix this in a follow-up PR.